Submitted By: Alexander E. Patrakov
Date: 2004-10-26
Initial Package Version: 5.4
Origin: Debian
Upstream Status: Upstream since 2004-03-20
Description: Fixed problem which prevented line-drawing characters from being
displayed in some 8-bit locales with wide-character version of ncurses.

diff -urN ncurses-5.4.orig/ncurses/curses.priv.h ncurses-5.4/ncurses/curses.priv.h
--- ncurses-5.4.orig/ncurses/curses.priv.h	2004-02-01 06:05:58.000000000 +0500
+++ ncurses-5.4/ncurses/curses.priv.h	2004-10-26 19:17:15.000000000 +0600
@@ -634,6 +634,7 @@
 #define isnac(ch)	(AttrOf(ch) & WA_NAC)
 #define if_WIDEC(code)  code
 #define Charable(ch)	((SP != 0 && SP->_posix_locale)			\
+			 || (AttrOf(ch) & A_ALTCHARSET)			\
 			 || (!isnac(ch) &&				\
 			     (ch).chars[1] == L'\0' &&			\
                              _nc_is_charable(CharOf(ch))))
diff -urN ncurses-5.4.orig/ncurses/tty/tty_update.c ncurses-5.4/ncurses/tty/tty_update.c
--- ncurses-5.4.orig/ncurses/tty/tty_update.c	2004-02-07 23:02:42.000000000 +0500
+++ ncurses-5.4/ncurses/tty/tty_update.c	2004-10-26 19:17:15.000000000 +0600
@@ -233,7 +233,6 @@
 	    chtype temp = UChar(SP->_acs_map[CharOfD(ch)]);
 	    if (temp != 0)
 		SetChar(my_ch, temp, attr);
-	    RemAttr(my_ch, A_ALTCHARSET);
 	}
 	ch = CHREF(my_ch);
     }
